### DISCUSSION TITLE
Tasmota controlled camera?

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16253] [Q&A] [answered]
Are there any PTZ cameras that can be controlled using tasmota? 
I've seen the little camera modules but I'm looking more for outdoor PTZ cameras where I could control the functions without using an app or browser.

Searching the net only leads to builder modules so I assume the answer is no.

**Comment by barbudor** [comment, marked as answer]
You can build yourself a PT camera using the ESP32-CAM module and 1 to 3 servos depending on which degrees of liberty you want.
For the Zoom, I'm not sure you can find a zoomable optic that can fit on the ESP32-CAM
Servos can be more or less controlled by PWM outputs although this is not perfect as Tasmota is targeting PWM for lights not for servos so there are some drawbacks.

Nevertheless the ESP32-CAM is crappy and I wouldn't used that for anything more than an toy.

**Comment by dac2020** [comment]
I prefer not to build one, just wondered if there were any that could be flashed with tasmota just for the controls, not the video output.
Mainly just don't want to use a mobile app to control PTZ.

Sounds like a no :). 

Thank you.

**Reply by barbudor** [reply]
I've never heard of a PTZ camera that would have an ESP32 inside

I'm also very angry against those recent cameras htat you can't use without the manufacturer app based on the manufacturer cloud.
There use to be cameras with embedded web control, no cloud connection and RTSP stream output. I had one but it failed beyond repair.
I you find one of those, I'll be happy to learn about

**Reply by sfromis** [reply]
For someone making a PTZ camera, selecting a bad camera like ESP32-CAM would be a weird mismatch of focus. For the low price, ESP32-CAM makes sense if bad quality is enough, but far from a "real" web/surveillance camera where PTZ would make sense.
Some of the cameras used with Raspberry Pi look decent, and likely have good use for a higher CPU performance than ESP32. And a higher price, of course.

**Reply by dac2020** [reply]
Just to be very clear... I am only looking for a camera that can be remotely controlled for the PTZ. 
The video would still be streamed as usual, what ever it's using.

I know there are lots of ONVIF ways to do it but wondered if there was something that could tie into this.

**Comment by dac2020** [comment]
Yes, that's the problem. All of the Chinese cameras come with their own 'custom' app so you have to sign up to as many sites as you own cameras if you don't buy the same. I'm sick and tired of signing up to unknown web services that barely work :).

I have a bunch of cameras, all have different web interfaces and apps so I started looking for cameras that might have alternate ways of controlling them. 

I'll share for sure if I come across something.

**Reply by 001Tycoon** [reply]
OpenIPC Firmware is available for many cameras. Look at https://openipc.org/

**Reply by dac2020** [reply]
Thank you for that. I'll take a look.

**Reply by dac2020** [reply]
I looked on ebay and was not able to find even one of the cameras listed in the hardware list.
https://openipc.github.io/wiki/en/guide-supported-devices.html

Is this a new project, have you used it?

**Reply by 001Tycoon** [reply]
This is firmware for the SoC on the board, not the camera manufacturers. You can use the tool https://github.com/OpenIPC/ipctool to check which SoC is installed.
A list of manufacturers https://openipc.github.io/wiki/en/hardware-board-manufacturers.html

I tested, the firmware works fine.

**Reply by dac2020** [reply]
I understand now. I thought those were camera models.
The problem with Chinese goods is someone starts off with some hardware and people start buying it. Soon you seen clones using different chips and now you end up buying clones without even knowing until people post their findings. 

I've gotten bitten by that many times.
